### DISCUSSION TITLE
Outloook stuck in loading state: TypeError when trying to parse query parameters

### DISCUSSION THREAD

**Discussion by Haraldson** [InteractionDesignFoundation/add-event-to-calendar-docs#47] [General]
Has anyone else encountered this and been able to get around it? I guess it happens due to a recent change on Microsoft’s side, and since this is so blackboxed, I’m not really sure how to proceed to get it working again.

**Comment by Haraldson** [comment]
It seems to work better when skipping the `/0` segment of the Office Outlook calendar URL.

**Comment by JJin-park** [comment]
Same issue to me. I think `/0` parameter is problem, when i removed that is work fine.
I think `/0` is calendar's index, right?
But i'm not sure removing `/0` is correct. What do you think about that? @Haraldson 

ps. Do you know MS Official document about this Outlook deeplink?
I can't find anything.

**Reply by Haraldson** [reply]
Removing `/0` broke it in Edge for us...

**Reply by JJin-park** [reply]
It's working on our env. 
I desperately finding Offical document...

**Reply by JJin-park** [reply]
Hi, I found the problem without `/0`.
Problem is only one account's calendar opened on multiple logined account session in same browser.

ex. A, B user are logined. When open link on B user's calendar, opening the A users's calendar.

**Reply by JJin-park** [reply]
@Haraldson @bgpearman 

Hi, I have a new update.
`https://outlook.office.com/calendar/{userPrincipalName}/deeplink/compose/?path=/calendar/action/compose&rru=addevent&startdt=2023-05-08T03:30:00.000Z&enddt=2023-05-08T04:00:00.000Z`

Use upn instead of `/0` is working. 
Above problem(multi account in same browser issue) give back Outlook Error `AccessDeniedException`
This process looks like right way to me. Opening only one user calendar on multiple login session without warning, it was big problem to me.

I don't remember how works when `\0` is work, i guess maybe gave back Outlook Error `AccessDeniedException`.

What do you think about change paremeter to upn.
Please, give me feedback.

**Reply by Haraldson** [reply]
What value does the `userPrincipalName` variable hold?

**Reply by jekuer** [reply]
Difficult imho, since you might not always know about the upn (e.g. dealing with anonymous users).

**Comment by bgpearman** [comment]
@JJin-park Did you find any official documentation?
Looks like the issue is related to multiple accounts with my testing too. The issue is resolved if I open a new incognito window, log in, and open an "add to calendar" link there

see also
https://techcommunity.microsoft.com/t5/outlook/https-outlook-office-com-calendar-0-deeplink-compose-fails-to/m-p/3816442

**Reply by JJin-park** [reply]
No, i gave up to find.
In my case, `incognito window` is not solution. I'm looking for other alternatives, but they don't seems to exist.
I considering create custom appointment view in worst.
Also, I open the Microsoft Case in our tenant, but they are notihing give to me any response yet.

I will monitoring the you shared link(thechcommuity), thanks.
Please, if you get any update for this issue, share to this dicussion.

**Comment by jekuer** [comment]
Hey guys,

I think I found the solution, but would love to gather more testing feedback from you!
To me, it seems like Microsoft changed the way they (re-)route traffic. Therefore, the deeplink workaround is no longer fully functional.
The following scheme (using the path query param directly) seems to work (taking your example, @JJin-park ):
`https://outlook.office.com/calendar/action/compose/?rru=addevent&startdt=2023-05-08T03:30:00.000Z&enddt=2023-05-08T04:00:00.000Z`

**Reply by bgpearman** [reply]
That works for me!

**Reply by bgpearman** [reply]
Not to derail the discussion, but I have recently started a similar issue with mailto links opening not opening in Outlook Web. Applying the equivalent change for calendar links for mailto links doesn't fix it:

Before v1 (bugged):
https://outlook.live.com/owa/?path=/mail/action/compose&to=mailto%3Aname1%40rapidtables.com%3Fcc%3Dname2%40rapidtables.com%26bcc%3Dname3%40rapidtables.com%26subject%3DThe%2520subject%2520of%2520the%2520email%26body%3DThe%2520body%2520of%2520the%2520email

Before v2 (bugged):
https://outlook.live.com/mail/0/deeplink/compose?to=mailto%3Aname1@rapidtables.com%3Fcc%3Dname2@rapidtables.com%26bcc%3Dname3@rapidtables.com%26subject%3DThe%2520subject%2520of%2520the%2520email%26body%3DThe%2520body%2520of%2520the%2520email

After (bugged):
https://outlook.live.com/mail/action/compose/?to=mailto%3Aname1@rapidtables.com%3Fcc%3Dname2@rapidtables.com%26bcc%3Dname3@rapidtables.com%26subject%3DThe%2520subject%2520of%2520the%2520email%26body%3DThe%2520body%2520of%2520the%2520email

**Reply by Haraldson** [reply]
That seems to work more reliably so far!

**Reply by JJin-park** [reply]
Thanks for your test, @jekuer.

It's same problem on multiple login session in one browser.
This is big deal in my env.

I think I cannot use `action` parameter. I prefer using `userPrincipalName`.
But I think it's good solution if not in my env.


There is the different between use `userPrincipalName` + `deeplink` and `action`.
The `deeplink` is opening only addevent calendar, but `action` is open addevent calendar on `outlook.office.com`.

**Reply by jekuer** [reply]
@JJin-park , just realized that the new scheme (`action`) is only working on desktop.
On mobile (without the app), only the old one is working (but with the known issues then).

**Reply by jekuer** [reply]
Seems like the Microsoft intern messed around with it again. Date and time information are also no longer picked up correctly.

**Reply by tagryn101** [reply]
I found this works for O365 Outlook, but haven't found a similar functional way for outlook.live.com : 

https://outlook.office365.com/owa/?path=/mail/action/compose&to=mailto%3Aname1@rapidtables.com%3Fcc%3Dname2@rapidtables.com%26bcc%3Dname3@rapidtables.com%26subject%3DThe%2520subject%2520of%2520the%2520email%26body%3DThe%2520body%2520of%2520the%2520email

**Comment by UNiXMIT** [comment]
> Hey guys,
> 
> 
> 
> I think I found the solution, but would love to gather more testing feedback from you!
> 
> To me, it seems like Microsoft changed the way they (re-)route traffic. Therefore, the deeplink workaround is no longer fully functional.
> 
> The following scheme (using the path query param directly) seems to work (taking your example, @JJin-park ):
> 
> `https://outlook.office.com/calendar/action/compose/?rru=addevent&startdt=2023-05-08T03:30:00.000Z&enddt=2023-05-08T04:00:00.000Z`

If I swap deeplink in the url for action in my email deeplinks it successfully opens outlook but with a blank email. So it may work for calendar but not for emails.

**Reply by bgpearman** [reply]
Yes I am having exactly the same issue.

**Comment by RichieMcCaff** [comment]
Was there any resolution to this?

We are still seeing the deeplink end on a blank screen and the outlook action/compose link does not work on mobile.

**Comment by alies-dev** [comment]
Re-tested the URL shapes in this thread against `outlook.live.com` on 2026-08-19.

**The documented deep link works, with or without the `/0` segment.** Both of these open the compose form with every field populated:

```
https://outlook.live.com/calendar/deeplink/compose?path=/calendar/action/compose&rru=addevent&startdt=2026-12-31T19:30:00Z&enddt=2026-12-31T22:30:00Z&subject=Test&body=b&location=L
https://outlook.live.com/calendar/0/deeplink/compose?path=/calendar/action/compose&rru=addevent&...
```

So on the consumer host the `/0` index is not the problem any more.

**The workaround posted here no longer works.** The `action/compose` form

```
https://outlook.live.com/calendar/action/compose/?rru=addevent&startdt=...&enddt=...&subject=...
```

lands on the calendar week view with no compose form at all. If anyone is still relying on it, it is worth re-checking.

What could not be tested here is `outlook.office.com`, which needs a work or school account. Since the original report and most of the follow-ups are about the Office 365 host, leaving this open.

One detail that may help whoever picks it up: the sign-in redirect preserves the deep link. The full compose URL is carried through the OAuth `state` parameter and comes back after authentication, so the parameters are not lost on the login round trip. Whatever breaks on Office 365 happens after that, inside the application. The useful thing to report is the exact console error plus whether the form appears at all.

**Comment by alies-dev** [comment]
Correction to my previous comment: I was wrong about the `action/compose` form.

It works, and the difference is the `path` parameter. This opens the inline compose panel on `outlook.live.com` with subject, body and location all populated, verified 2026-08-19:

```
https://outlook.live.com/calendar/action/compose?path=/calendar/action/compose&rru=addevent&startdt=2026-12-31T19:30:00Z&enddt=2026-12-31T22:30:00Z&subject=Test&body=desc&location=addr
```

What fails is the form without `path`, which is how the workaround was posted in this thread:

```
https://outlook.live.com/calendar/action/compose/?rru=addevent&startdt=...&enddt=...&subject=...
```

That one lands on the calendar week view with no compose panel. So `path=/calendar/action/compose` is required regardless of which of the two paths you use, and `spatie/calendar-links` (which emits the first form) is fine.

The rest of the earlier comment stands: `/calendar/deeplink/compose` works with and without the `/0` segment, and `outlook.office.com` remains untested here.
